tests/baselines/reference/typeGuardIntersectionTypes.types

```diff
@@ -241,16 +241,16 @@ function identifyBeast(beast: Beast) {
 >                : ^^^^^^^
 >hasWings : (x: Beast) => x is Winged
 >         : ^ ^^     ^^^^^^^^^^^^^^^^
->beast : Legged
->      : ^^^^^^
+>beast : Beast & Legged

```

Comment:
This is essentially the same test case as the added one. Without the change, we can't access `beast.wings` here (even though it's permitted before narrowing). 

Regardless of that, I find it useful to include the @danvk's test case here as I could include the ticket reference in a clean manner above it.